src/main/java/uk/gov/justice/probation/courtcaseservice/service/UserAgnosticOffenderService.java

```diff
@@ -51,8 +51,12 @@ public Optional<OffenderEntity> updateOffenderProbationStatus(String crn) {
     private OffenderEntity updateProbationStatusDetails(ProbationStatusDetail probationStatusDetail, OffenderEntity offender) {
         offender.setProbationStatus(OffenderProbationStatus.of(probationStatusDetail.getStatus()));
         offender.setPreviouslyKnownTerminationDate(probationStatusDetail.getPreviouslyKnownTerminationDate());
-        offender.setBreach(probationStatusDetail.getInBreach());
-        offender.setAwaitingPsr(probationStatusDetail.getAwaitingPsr());
+        if (probationStatusDetail.getInBreach() != null) {
```

Comment:
We generally prefer `Optional.ofNullable(probationStatusDetail.getInBreach()).orElse(false)` rather than if statements for null safety

Reply:
I initially implemented `Optional.ofNullable(probationStatusDetail.getInBreach()).orElse(offeder.isBreach())` but changed to avoid accessing the same field unnecessarily.

Reply:
Optional.ifPresent(..) better choice here satisfying both

Reply:
Yup agree.